samples/word/25-paragraph/onadded-event.yaml

```diff
@@ -0,0 +1,128 @@
+
+        let eventContext;
+
+        async function registerEventHandler() {
+          // Registers the onParagraphAdded event handler on the document.
+          await Word.run(async (context) => {
+            eventContext = context.document.onParagraphAdded.add(paragraphAdded);
+            await context.sync();
+
+            console.log("Added event handler for when paragraphs are added.");
+          });
+        }
+
+        async function paragraphAdded(event: Word.ParagraphAddedEventArgs) {
+          await Word.run(async (context) => {
+            console.log(`${event.type} event detected. IDs of paragraphs that were added:`);
+            console.log(event.uniqueLocalIds);
+          });
+        }
+
+        async function getParagraphById() {
+          await Word.run(async (context) => {
+            const paragraphId = $("#paragraph-id").val() as string;
+            const paragraph = context.document.getParagraphByUniqueLocalId(paragraphId);
+            paragraph.load();
+            await paragraph.context.sync();
+
+            console.log(paragraph);
+          });
+        }
+
+        async function insertParagraphs() {
+          // Inserts two paragraphs within the document body.
+          await Word.run(async (context) => {
+            const paragraphCount = 2;
+            for (let i = 0; i < paragraphCount; i++) {
+              context.document.body.insertParagraph(`Paragraph Test ${i + 1}`, "End");
+            }
+
+            console.log("Paragraphs inserted: " + paragraphCount);
+            await context.sync();
+          });
+        }
+
+        async function deregisterEventHandler() {
+          await Word.run(eventContext.context, async (context) => {
+            eventContext.remove();
+            await context.sync();
+          });
+
+          eventContext = null;
+          console.log("Removed event handler that was tracking when paragraphs are added.");
+        }
+
+        /** Default helper for invoking an action and handling errors. */
+        async function tryCatch(callback) {
+          try {
+            await callback();
+          } catch (error) {
+            // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
+            console.error(error);
+          }
+        }
+    language: typescript
+template:
+    content: |-
+        <section class="ms-font-m">
+            This sample demonstrates how to use the onAdded event with paragraphs.
+        </section>
+
+        <section class="samples ms-font-m">
+            <h3>Try it out</h3>
+            <button id="register-event-handler" class="ms-Button">
+            <span class="ms-Button-label">Register event handler</span>
+          </button>
+            <button id="insert-paragraphs" class="ms-Button">
+              <span class="ms-Button-label">Insert paragraphs</span>
+          </button>
+            <p>
+                <label style="margin-left: 20px">Paragraph ID:</label>
```

Comment:
Suggestion: Consider adding text before the paragraph field to notify the user to copy the paragraph ID from the console.

Reply:
Added